xpdacq/yamlclass.py

```diff
@@ -0,0 +1,64 @@
+##############################################################################
+#
+# xpdacq            by Billinge Group
+#                   Simon J. L. Billinge sb2896@columbia.edu
+#                   (c) 2016 trustees of Columbia University in the City
+#                   of
+#                        New York.
+#                   All rights reserved
+#
+# File coded by:    Timothy Liu
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE.txt for license information.
+#
+##############################################################################
+
+import os
+import yaml
+
+class YamlClass:
+    """
+    special class automatically yamlize user-defined attributes
+    if they are updated
+    """
+
+    def __init__(self, internal_dict=None):
+        # dict stores values of valid attributes
+        if internal_dict is None:
+            internal_dict = {}
```

Comment:
please see: 
http://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

It is a convention to avoid confusion on default values.